### ISSUE TITLE
debugger taking steps on stuck configuration

### ISSUE THREAD

**Issue by bmmoore** [kframework/k-legacy#1583]
When a test program gets unexpectedly stuck, I expected I could use the debugger to find how it got there by running commands like

```
step -s 10000
back-step
```

Instead, the debugger seems to consider trying to take a step on a stuck configuration as actually making a step of progress, which I need to back-step over.
I've assigned this to @msaxena2 under the assumption it's in the debugger, because I've seen it both with the Java backend and the O'Caml backend, but I haven't checked if it's a shared bug.

This can be reproduced with the following definition as stuck.k

```
require "domains.k"
module STUCK-SYNTAX
syntax PGM ::= "stuck"
endmodule

module STUCK
imports MAP
imports STUCK-SYNTAX
configuration <k>$PGM:K</k>
endmodule
```

compiled with `kompile --kore stuck.k`, and run on a test program `test.stuck` consisting just of `stuck`, with the  following input demonstrating that it thinks "step" actually made progress (blank lines removes):

```
$ krun --kore --debugger test.stuck
Command > back-step
Invalid Operation
Command > step
<k> stuck </k>
Command > back-step
<k> stuck </k>
```

   Command >

**Comment by msaxena2** [comment]
I'm aware of this problem, and you're right it's in the debugger. Will be fixing it soon in the next Pull Request.

**Comment by msaxena2** [comment]
#1555 Fixes this Issue. 
Behavior of the debugger on the same program now:

KDebug> s
Attempted 1 step(s). Took 0 steps(s).
Final State Reached
KDebug> b
Already at Start State, Cannot take steps.
KDebug> s 10
Attempted 10 step(s). Took 0 steps(s).
Final State Reached
KDebug> b 10
Number of Configuration(s) is 1. Step Count to go back must be in range [0, 1)

And yes, s and b can be used to called step and back-step now (along with step and back-step, of course).

**Comment by bmmoore** [comment]
Sounds like it's fixed, but it looks like `b` is unsymmetrically giving up without trying to take any steps, rather than backing up to an initial state. Is there is reason to make them different? (I'm not sure I'd even prefer you to change it, reportiing the actual count seems fairly reasonable too, unless b with a big number is the best way of restarting).

**Comment by msaxena2** [comment]
When it comes to stepping back, the debugger already knows how many steps it can take before hitting an initial state (unlike stepping forward, where it doesn't know the number of steps it can take before reaching the final state), hence it regards an instruction to take more steps than permissible as invalid, and doesn't do anything at all. As far as restarting is concerned, there are many ways of doing that (for e.g., you can instruct it to jump to the initial state itself, which will effectively restart the debugging session).

**Comment by bmmoore** [comment]
I see there was a little back and forth merging the pull request, but it's fixed now (I tested on master as of 425f4a45ba5e1d03db46b6e5bc4a61e7b11f226b).
